src/sentry/static/sentry/app/components/events/interfaces/threads/threadSelector/getThreadStacktrace.tsx

```diff
@@ -1,24 +1,47 @@
-import {ExceptionValue} from 'app/types';
+import {ExceptionType} from 'app/types';
 import {Event} from 'app/types/event';
 import {Thread} from 'app/types/events';
 
 import getThreadException from './getThreadException';
 
-function getThreadStacktrace(thread: Thread, event: Event, raw: boolean) {
-  const exc = getThreadException(thread, event);
-  if (exc) {
-    let rv: ExceptionValue['stacktrace'] | undefined = undefined;
-
-    for (const singleExc of exc.values) {
-      if (singleExc.threadId === thread.id) {
-        rv = singleExc.stacktrace;
-        if (raw && singleExc.rawStacktrace) {
-          rv = singleExc.rawStacktrace;
-        }
-      }
+function getThreadStacktrace(
+  event: Event,
+  raw: boolean,
+  thread?: Thread,
+  exception?: Required<ExceptionType>
+) {
+  if (!thread) {
+    return undefined;
+  }
+
+  const threadException = exception ?? getThreadException(event, thread);
+
+  if (threadException) {
+    const threadExceptionValues = threadException.values;
+
+    const matchedStacktraceAndExceptionThread = threadExceptionValues.find(
+      threadExceptionValue =>
+        threadExceptionValue.threadId === thread.id && threadExceptionValue.stacktrace
+    );
+
+    // When a exception value matches the threadId, it means that the stack traces are the same
+    // and we don't want to display the same info 2x in the UI
+    if (matchedStacktraceAndExceptionThread) {
+      return undefined;
     }
 
-    return rv;
+    // In the getThreadException function, when threadIds don't match the activeThread id and the activeThread is 'crashed' and
+    // when at least one exception stack trace is found in the exception values, the function returns the exception data.
+    // When this happen undefined has to be returned here
+    if (thread.crashed) {
+      const exceptionHasAtLeastOneStacktrace = !!threadExceptionValues.find(
+        threadExceptionValue => threadExceptionValue.stacktrace
+      );
+
+      if (!!exceptionHasAtLeastOneStacktrace) {
+        return undefined;
```

Comment:
I don't understand this. Isn't this exactly the case where we want to show the exception within the thread? The one we are trying to fix?

Reply:
No. The problem we are trying to solve is that when an event has threads without stack traces but exceptions. the UI should display the exceptions.

Reply:
oh right, this is getThreadStacktrace 👍